### PR TITLE
Ticket4992 raise invalid period exception

### PR DESCRIPTION
**Pull request by Alexandru-Dascalu**
### Description of work

Added system tests to test the genie python commands for changing the period and the total number of periods for both normal and for invalid input. 

### Ticket

https://github.com/ISISComputingGroup/IBEX/issues/4992

### Acceptance criteria

Newly added system tests pass on dev machines and also on our build server.

### System tests

2 for each of the two methods, one for normal input and for when genie python now throws an error. The tests for change_period are parameterized and use Boundary Value Analysis.

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev)?

### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: test_genie_python_dae.py [hunk 1]
```diff
@@ -310,6 +310,53 @@ def test_GIVEN_change_tables_called_WHEN_filenames_are_not_raw_strings_and_with_
...
+        g.change_number_soft_periods(30)
+        sleep(10)
+        self.assertEqual(g.get_number_periods(), 30)
+
+        self.assertRaises(IOError, g.change_number_soft_periods, 1000000)
+        self.assertEqual(g.get_number_periods(), 30)
+
+        set_genie_python_raises_exceptions(False)
+
+    @parameterized.expand(parameterized_list([1, 2, 6, 9, 10]))
```
Comment (RaiBishal) [inline test_genie_python_dae.py]: we seem to be not using the 0th indexed tuple value from each tuple. Why not just pass a list of lists?

File: test_genie_python_dae.py [hunk 2]
```diff
@@ -310,6 +310,53 @@ def test_GIVEN_change_tables_called_WHEN_filenames_are_not_raw_strings_and_with_
...
+        g.change_number_soft_periods(10)
+        sleep(10)
+        self.assertEqual(g.get_number_periods(), 10)
+
+        g.change_period(new_period)
+        self.assertEqual(g.get_period(), new_period)
+
+        set_genie_python_raises_exceptions(False)
+
+    @parameterized.expand(parameterized_list([-1, 0, 11, 12]))
```
Comment (RaiBishal) [inline test_genie_python_dae.py]: Same as above

File: test_genie_python_dae.py (hunk 1 above)
Comment (Alexandru-Dascalu) [reply test_genie_python_dae.py]: It is just easier this way and looks less messy

File: test_genie_python_dae.py (hunk 2 above)
Comment (Alexandru-Dascalu) [reply test_genie_python_dae.py]: It is just easier this way and looks less messy

File: test_genie_python_dae.py [hunk 3]
```diff
@@ -310,6 +310,53 @@ def test_GIVEN_change_tables_called_WHEN_filenames_are_not_raw_strings_and_with_
             spectra=spectra
         )
 
+    def test_GIVEN_change_number_soft_periods_called_WHEN_new_value_normal_THEN_change_successful(self):
+        set_genie_python_raises_exceptions(True)
+
+        g.change_number_soft_periods(30)
+        sleep(10)
```
[the remark this answers is not in the document]
Comment (Alexandru-Dascalu) [reply test_genie_python_dae.py]: Is TOO_MANY_PERIODS_FOR_DAE a good name?

File: test_genie_python_dae.py [hunk 4]
```diff
@@ -310,6 +310,53 @@ def test_GIVEN_change_tables_called_WHEN_filenames_are_not_raw_strings_and_with_
...
+        set_genie_python_raises_exceptions(False)
+
+    def test_GIVEN_change_number_soft_periods_called_WHEN_new_value_too_big_for_DAE_hardware_THEN_raise_exception_to_console(self):
+        set_genie_python_raises_exceptions(True)
+
+        g.change_number_soft_periods(30)
+        sleep(10)
+        self.assertEqual(g.get_number_periods(), 30)
+
+        self.assertRaises(IOError, g.change_number_soft_periods, 1000000)
```
[the remark this answers is not in the document]
Comment (Alexandru-Dascalu) [reply test_genie_python_dae.py]: Is TOO_MANY_PERIODS_FOR_DAE a good name?
